lib/client.js

```diff
@@ -119,7 +119,7 @@ Client.prototype._request = function (method, url, opts, cb) {
       err.statusCode = res.statusCode;
       err.body = body;
 
-      return cb(err);
+      return cb(err, res);
```

Comment:
We need to return `res` even in the error case so we can check the `cache-control` header